### DISCUSSION TITLE
Add interactive mode to the GMT

### DISCUSSION THREAD

**Discussion by ribalba** [green-coding-solutions/green-metrics-tool#607] [Ideas]
Following discussions with @ArneTR and @davidkopp we came up with the idea of adding an interactive monitoring mode to the Green Metrics Tool. The current problem is that you have to specify all the docker container beforehand and thus dynamic scaling or measuring complex scenarios is tricky. The idea is that we add a mode in which all containers on a system are benchmarked while the GMT is running. The metrics providers would have to scan for new containers on every tick and output the container specific data. This would enable us to measure complex scenarios with containers starting and exiting. Of course the boot phase of the contains might be difficult to measure in each instance as other contains might already be generating load. But with this approach we can get comparable runs.

**Comment by davidkopp** [comment]
For reference, a PR exists, currently in draft state, that adds a monitor mode to GMT: https://github.com/green-coding-berlin/green-metrics-tool/pull/630
